### DISCUSSION TITLE
How to call removeEDNSOptionFromOPT in dnsdist.conf ?

### DISCUSSION THREAD

**Discussion by hi-glenn** [PowerDNS/pdns#14769] [Q&A]
How to call `int removeEDNSOptionFromOPT(char* optStart, size_t* optLen, const uint16_t optionCodeToRemove)` function by lua in dnsdist.conf

My dns server is too old to support edns-cookie. I want to use dnsdist as a proxy to remove edns-cookie on query.

How should I write the lua script ?

**Comment by rgacogne** [comment]
There is currently no binding to make this function available to Lua, so it's not possible. You could patch dnsdist yourself, to add a Lua binding for this function, but you would need to be careful: `removeEDNSOptionFromOPT` only works as expected if there is no RR after the OPT one, which is usually the case but might not be true with `TSIG`, for example.
I would personally rather investigate upgrading/replacing a DNS server that chokes on DNS cookies (which should not be the case, a server that does not understand an EDNS option should just ignore it) than working around it, but I guess that might not be an option for you.
